unix/src/machine_stats/modules/proc_stats.py

```diff
@@ -58,7 +58,12 @@ def parse_status(process_path):
     # Note : This call requires root level access to be able to follow
     # all symlinks. Otherwise some processes will be identified as
     # /proc/2138/exe
-    path, name = str(Path(process_path + "/exe").resolve()).rsplit("/", 1)
+
+    try:
```

Comment:
My best guess is that a broken symlink indicates this:
```
              In a multithreaded process, the contents of this symbolic
              link are not available if the main thread has already
              terminated
```

And then it's not a process that's actually running (it's at best a subprocess?) and then shouldn't be reported?

Reply:
thanks for taking a look!

the process that caused this issue was `kthreadd` (pid `2`) its the 'kernel thread'. its special and part of the kernel to run threads within the kernel, its not a regular user process. likely why the symlink does not resolve to a running process at a filepath on the filesystem, the process is inside the _the kernel_.

The fact is that sym links are 'broken' be design in 'proc' - https://unix.stackexchange.com/questions/197854/how-does-the-proc-pid-exe-symlink-differ-from-ordinary-symlinks

they are not expected to resolve to file paths and behave how symlinks do for 'real' files on the filesystem.

we can add errors for certain cases if we need to. but until then, this is an improvement from today, with some data returned being returned here.


I'll release the fix?

Reply:
> they are not expected to resolve to file paths and behave how symlinks do for 'real' files on the filesystem.

The docs say they are supposed to resolve to a path and only break in a specific circumstance.

>we can add errors for certain cases if we need to. but until then, this is an improvement from today, with some data returned being returned here.
> 
> I'll release the fix?

I think there is a potential new bug introduced by this fix: if a user doesn't have their permissions right they won't be able to read about processes started by other users. I call this a potential bug because it depends on what users expect out process stats? Does it guarantee to get all running processes on the machine? Or all processes running for that user?

I was under the assumption it was all, but if it's the latter feel free to release.